src/main/java/org/apache/mesos/hdfs/config/SchedulerConf.java

```diff
@@ -1,10 +1,28 @@
 package org.apache.mesos.hdfs.config;
 
+import com.google.inject.Singleton;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
 
-public class SchedulerConf extends MainConf {
-  public SchedulerConf(Configuration conf, int configServerPort) {
-    super(conf, configServerPort);
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Properties;
+
+@Singleton
+public class SchedulerConf extends Configured {
+
+  public SchedulerConf(Configuration conf) {
+    setConf(conf);
+  }
+
+  public SchedulerConf() {
+    Properties props = System.getProperties();
```

Comment:
It is clear now, although I also would like to think about further clean up in the future. Thanks for describing your use case. At the very least, let's merge these 2 values into one property (instead of sitePath and configPath, let's just have configPath to avoid adding configurations we are not yet using and to make the code less ugly). Also, can we add a comment?